ktor-http/jvm/src/io/ktor/http/content/BlockingBridge.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.http.content
+
+import kotlinx.coroutines.*
+import java.lang.reflect.*
+
+private val isParkingAllowedFunction: Method? by lazy {
+    try {
+        Class.forName("io.ktor.utils.io.jvm.javaio.PollersKt")
```

Comment:
why using reflection here?

Reply:
Because we can't invoke the function directly: it is made internal to avoid public API change and we are calling it via reflection. If the function is missing (like we have some kind jar hell), then no invocation happens and we fall back to the old behaviour. In 1.5.x we'll be able to make functions public and remove reflection magic.

Reply:
So it's a trick